libsmtutil/genz3wrapper.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+import sys
+import re
+
+# Patterns to match Z3 API entry point definitions.
+def_pat = re.compile(" *def_API(.*)")
+extradef_pat = re.compile(" *extra_API(.*)")
+# Pattern to extract name and arguments from the above.
+def_args_pat = re.compile("\('(.*)'[^\(\)]*\((.*)\)\s*\)")
```

Comment:
```suggestion
def_args_pat = re.compile("\('([^']*)'[^\(\)]*\((.*)\)\s*\)")
```
But even with this it will be wonky on unexpected input like strings containing `\'` or two parameters containing parentheses.

Some validations that the `def_API` line actually has the format we expect (and raising an error if it does not) is needed here in my opinion.

Reply:
It is all validated :-). If any of those lines falls through here or a name is parsed incorrectly, then we get undefined-symbol linker errors. If the arguments are parsed incorrectly, then it's compile-time errors due to the type in ``reinterpret_cast<decltype(&<functionName>)>`` not being deducible later.
Also Z3 generates it's python bindings from those lines, so they are very predictably formatted.

Reply:
That being said, excluding the ``'`` there also doesn't hurt :-).